Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp

```diff
@@ -579,15 +592,35 @@ ThrowCompletionOr<Value> to_relative_temporal_object(VM& vm, Object const& optio
         // a. If value has either an [[InitializedTemporalDate]] or [[InitializedTemporalZonedDateTime]] internal slot, then
         if (is<PlainDate>(value_object) || is<ZonedDateTime>(value_object)) {
             // i. Return value.
-            return value;
+            if (is<ZonedDateTime>(value_object)) {
```

Comment:
Seems like we can update the spec comments and follow the same if/else chain:
```
a. If value has an [[InitializedTemporalZonedDateTime]] internal slot, then
    i. Let timeZoneRec be ? CreateTimeZoneMethodsRecord(value.[[TimeZone]], « get-offset-nanoseconds-for, get-possible-instants-for »).
    ii. Return the Record { [[PlainRelativeTo]]: undefined, [[ZonedRelativeTo]]: value, [[TimeZoneRec]]: timeZoneRec }.
b. If value has an [[InitializedTemporalDate]] internal slot, then
    i. Return the Record { [[PlainRelativeTo]]: value, [[ZonedRelativeTo]]: undefined, [[TimeZoneRec]]: undefined }.
c. If value has an [[InitializedTemporalDateTime]] internal slot, then
    i. Let plainDate be ! CreateTemporalDate(value.[[ISOYear]], value.[[ISOMonth]], value.[[ISODay]], value.[[Calendar]]).
    ii. Return the Record { [[PlainRelativeTo]]: plainDate, [[ZonedRelativeTo]]: undefined, [[TimeZoneRec]]: undefined }.
```

Reply:
Right, I actually did do it that way originally, but because this AO is in a weird half-updated state I had ended up undoing that.

I've updated it now though to closer follow the newer version of the spec :^)